### **âŒ PROHIBIDO:**

1.  **Comandos desde subdirectorios:**
    ```bash
    # âŒ NUNCA HACER ESTO:
    cd Demo/apps/superapp-unified/ && npm run dev
    cd backend/ && npm run dev
    ```

2.  **MÃºltiples procesos simultÃ¡neos:**
    ```bash
    # âŒ NUNCA ejecutar mÃºltiples instancias
    npm run dev  # Ya ejecutÃ¡ndose
    npm run dev  # âŒ Segunda instancia causa conflictos
    ```

3.  **ğŸ†• Iniciar backend sin dependencias crÃ­ticas:**
    ```bash
    # âŒ NUNCA iniciar backend sin verificar PostgreSQL Y Redis
    npm run dev:backend  # Sin verificar brew services list | grep postgresql
    npm run dev:backend  # Sin verificar brew services list | grep redis
    ```

4.  **Mezclar archivos backend/frontend:**
    ```bash
    # âŒ NUNCA poner archivos React en:
    ./src/components/          # âŒ Es directorio de backend
    ./src/pages/              # âŒ Es directorio de backend
    ```

5.  **Ignorar limpieza de procesos:**
    ```bash
    # âŒ NUNCA iniciar sin limpiar:
    npm run dev  # Sin pkill previo causa conflictos
    ```

6.  **Usar solo Turborepo global:**
    ```bash
    # âŒ EVITAR: Solo instalaciÃ³n global (causa advertencias)
    # WARNING: No locally installed `turbo` found
    ```

7.  **ğŸ†• Importaciones lazy loading incorrectas:**
    ```bash
    # âŒ NUNCA usar nombres que no coincidan con archivos reales
    import('../pages/UPlayPage')    # âŒ Si UPlay.tsx es el archivo real
    import('../pages/ProfilePage')  # âŒ Si Profile.tsx es el archivo real
    ```

8.  **ğŸ†• ConfiguraciÃ³n incorrecta del compilador NestJS para Prisma:**
    ```bash
    # âŒ NUNCA olvidar configurar nest-cli.json para copiar generated/prisma
    # Asegurarse de que el outDir sea 'dist'
    # { "include": "generated/prisma/**/*", "outDir": "dist" }
    ```

9.  **ğŸ†• Enviar comandos de inicio a segundo plano durante la depuraciÃ³n:**
    ```bash
    # âŒ NUNCA enviar a segundo plano npm run dev o cd backend/ && npm run dev
    # cuando se estÃ¡n diagnosticando problemas de inicio.
    # Siempre observar la salida completa.
    ```

### **âœ… OBLIGATORIO:**

1.  **ğŸ†• Siempre verificar dependencias crÃ­ticas primero:**
    ```bash
    brew services list | grep postgresql  # Verificar PostgreSQL
    brew services list | grep redis       # Verificar Redis
    brew services start postgresql@15     # Iniciar PostgreSQL si es necesario
    brew services start redis            # Iniciar Redis si es necesario
    lsof -i :5432                        # Verificar conectividad PostgreSQL
    lsof -i :6379                        # Verificar conectividad Redis
    ```

2.  **Siempre verificar ubicaciÃ³n:**
    ```bash
    pwd  # Debe mostrar: /Users/kevinp/Movies/GAMIFIER-copy
    ```

3.  **Siempre limpiar procesos:**
    ```bash
    pkill -f "vite" && pkill -f "npm run dev" && pkill -f "turbo"
    ```

4.  **Instalar Turborepo localmente:**
    ```bash
    npm install turbo --save-dev --legacy-peer-deps
    npm ls turbo  # Verificar instalaciÃ³n local
    ```

5.  **Siempre usar workspace sintaxis:**
    ```bash
    npm run <script> --workspace=<nombre_del_paquete>
    ```

6.  **Siempre mantener separaciÃ³n arquitectÃ³nica:**
    ```bash
    # Backend: ./src/ - SOLO NestJS
    # Frontend: Demo/apps/superapp-unified/src/ - SOLO React
    ```

7.  **ğŸ†• Verificar importaciones lazy loading:**
    ```bash
    ls Demo/apps/superapp-unified/src/pages/  # Verificar archivos reales
    # Mapear correctamente: UPlay.tsx â†’ import('../pages/UPlay')
    ```

8.  **ğŸ†• Observar la salida completa de los comandos de inicio:**
    ```bash
    # Al depurar problemas de inicio del backend, siempre ejecutar los comandos
    # en primer plano para ver los logs y errores completos.
    ``` 
